.github/workflows/codeql.yml

```diff
@@ -5,22 +5,40 @@ on:
     branches: ['main', '*/main']
 
   pull_request:
+    types: [opened, synchronize, labeled]
 
 jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
+    # We run codeql only if it's:
+    #   1) pull request not from a fork (ie. internal PR), or
+    #   2) pull request from a fork (ie. external PR) that was added "run-codeql" label
+    if: |
+      (github.event.pull_request.head.repo.full_name == github.repository) ||
+      (github.event.action == 'labeled' && github.event.label.name == 'run-codeql')
     permissions:
       actions: read
       contents: read
       security-events: write
+      pull-requests: write # used to remove label
 
     strategy:
       fail-fast: false
       matrix:
         language: [javascript]
 
     steps:
+      - name: Remove run-codeql label, if applicable
```

Comment:
curious, why do we remove the label first instead of last?

Reply:
Good question, I had it run first so that labels can be re-added if someone adds a new commit before previous run adds.